lib/Backend/PreLowerPeeps.cpp

```diff
@@ -98,6 +98,14 @@ IR::Instr *Lowerer::PeepShl(IR::Instr *instrShl)
     instrDef->Remove();
 
     IntConstType oldValue = src2->AsIntConstOpnd()->GetValue();
+
+    // Left shift operator (<<) on arm32 is implemented by LSL which doesn't discard bits beyond lowerest 5-bit.
```

Comment:
@ThomsonTan AFAIK shift range is specified on a 5 bits field in the instruction. 5 bits give you the range of 0-31 (32 different values) Isn't it enough to shift all the bits on a 32 bit value? Could you please elaborate a bit more?